src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/IRequestHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+#nullable enable
+
+namespace CommonLanguageServerProtocol.Framework;
+
+/// <summary>
+/// Top level type for LSP request handler.
+/// </summary>
+public interface IRequestHandler<RequestContextType>
```

Comment:
Not sure we need the `<RequestContextType>` here or am I missing something?

Reply:
Ah good eye, we can remove it from this level.